index.d.ts

```diff
@@ -0,0 +1,456 @@
+declare module 'oidc-provider' {
+    import { Request, Response, Handler } from 'express';
+    import Koa = require('koa');
+    import Router = require('koa-router');
+    import { IncomingMessage, ServerResponse } from 'http';
+    import EventEmitter = require('events');
+    import { JWK } from 'node-jose';
+
+    type SubjectType = 'public' | 'parwise';
+
+    interface IClient {
```

Comment:
awful lot of metadata properties missing, please combine the following two sources

1) generic metadata https://github.com/panva/node-oidc-provider/blob/v5.1.2/lib/consts/client_attributes.js#L1-L28
2) feature specific metadata (`RECOGNIZED_METADATA.push`) from https://github.com/panva/node-oidc-provider/blob/v5.1.2/lib/helpers/client_schema.js#L53-L126

Reply:
Can you provide suggested types for them? Then I'd love to comb through the files and include the ones that aren't already there.

Reply:
I ended up trying to put all of them in in an additional commit. Take a look and let me know if it seems sensible.